scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,22 @@
+# 0005 - Adapt Scanner V4 NVD CVSS data updating pipeline to integrate with NVD CVE Api
+
+- **Author(s):** Yi Li <yli3@redhat.com>
+- **Created:** [2023-10-10]
+
+## Context
+
+The National Vulnerability Database (NVD) has announced the retirement of its JSON feeds by December 15, 2023. This creates the need to change our existing CVSS data updater pipeline for Scanner V4, which currently relies on these JSON feeds.
+
+## Decision
+
+The NVD CVSS Updater will retrieve CVSS data from Google storage at a configurable interval. It will make a single HTTP call for each NVD data bundle categorized by CVSS V3 severity. As an illustration, all CVE data with a CVSS V3 severity of 'low' will be archived into 'severity-low.zip' in Google storage and then being fetched by the updater.
+
+A GitHub workflow will keep the NVD data bundle in the Google Storage up-to-date. This Github workflow send http request to NVD CVE api to get data based on each level of CVSS V3 severities. Such as "curl 'https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0' > severity-critical-0.json".  Per the NVD's documentation, the NVD CVE API omits CVSS v3 vector strings with a 'NONE' severity. This ensures we only obtain data bearing valid CVSS V3 metrics. For JSON files of a singular severity, we compress them into a single zip file, such as severity-low.zip.
+
+This updater will download zip files corresponding to four distinct severity levels. Then it will write each json file from these zipped archives to a single zip file in Central's file system. We will not need to do any json parsing in the updater. This compressed file is typically around 50 MB in size.
```

Comment:
Just to confirm:

> This updater will download zip files corresponding to four distinct severity levels.

"Low", "Medium", "High", and "None"? I understood that the API filters scores with `severity == "None"`:

 > Per the NVD's documentation, the NVD CVE API omits CVSS v3 vector strings with a 'NONE' severity.

Reply:
"Low", "Medium", "High", and "Critical"